### PR TITLE
validate required components

### PR DESCRIPTION
**Pull request by Ghostvv**
**Proposed changes**:
- add validation of required components
- part of https://github.com/RasaHQ/research/issues/62

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/components.py [hunk 1]
```diff
@@ -51,13 +51,45 @@ def validate_requirements(component_names: List[Text]) -> None:
...
+
+
+def validate_required_components(pipeline: List["Component"]) -> None:
+    unique_component_names = set()
+    for component in pipeline:
+        unique_component_names.add(component.name)
+        if not set(component.required_components).issubset(unique_component_names):
+            raise Exception(
+                f"'{component.name}' requires {component.required_components}. "
+                f"Add required components to the pipeline"
```
Comment (tabergma) [inline rasa/nlu/components.py]:
```suggestion
                f"Add required components to the pipeline."
```

File: rasa/nlu/components.py [hunk 2]
```diff
@@ -223,6 +255,9 @@ def name(self):
     # provided properties from the previous components.
     requires = []
 
+    # Which components are required by this component
```
Comment (tabergma) [inline rasa/nlu/components.py]:
```suggestion
    # Which components are required by this component. Listed components should appear before 
    # the component itself in the pipeline.
```

File: rasa/nlu/components.py [hunk 3]
```diff
@@ -51,13 +51,41 @@ def validate_requirements(component_names: List[Text]) -> None:
         )
 
 
+def validate_tokenizers(pipeline: List["Component"]) -> None:
```
Comment (tabergma) [inline rasa/nlu/components.py]: Should we add a doc string? (same for the other method)

**Comment by Ghostvv** [comment]
Trying to find a place where to add it. Do we have tests for other validation functions?

**Comment by tabergma** [comment]
Should fit into `tests.nlu.test_config`

File: rasa/nlu/components.py [hunk 4]
```diff
@@ -51,17 +51,46 @@ def validate_requirements(component_names: List[Text]) -> None:
...
+            tokenizer_names.append(component.name)
+
+    if len(tokenizer_names) > 1:
+        raise InvalidConfigError(
+            f"More then one tokenizer is used: {tokenizer_names}. "
+            f"You can use only one tokenizer."
+        )
+
+
+def validate_required_components(pipeline: List["Component"]) -> None:
```
Comment (tmbo) [inline rasa/nlu/components.py]: did this in the other pr, but should probably go here:
why do we need this? should this not be covered by a components `requires` property

Comment (tabergma) [reply rasa/nlu/components.py]: Partly, some components, such as `ConveRTFeaturizer`, require `tokens` from the `ConveRTTokenizer`. Just saying the `ConveRTFeaturizer` requires `tokens` would not be enough.

**Pull request merged**